src/types/yargs.d.ts

```diff
@@ -0,0 +1,13 @@
+import { Arguments, Options } from "yargs";
+
+interface Argument extends Arguments{
```

Comment:
Let's rename this to `YargsArguments` (or, if you're feeling funny, `YargsArgs`) for great clarity.

We also need a dynamic way to indicate which properties will be present, because in most cases in practice, not all of these properties will be set. There might be a way to do this with generics. We should talk about it and do some research 🤔

Reply:
of course im feeling funny